### PR TITLE
fix the sendLast which might cause hung

### PR DESCRIPTION
**Pull request by dongdongyang33**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue MO-Cloud#643

## What this PR does / why we need it:
If some of a set of right join did not go into the sendLast() function when clean-up,
the pipeline might be hung.

### CODE REVIEW

**Comment by matrix-meow** [comment]
@dongdongyang33 Thanks for your contributions!

The pull request fixes an issue where the pipeline might hang if some of the right join set did not go into the sendLast() function when clean-up. The changes include adding a new boolean field `handledLast` to the container struct, setting it to false in the Prepare() function, and setting it to true in the sendLast() function. The Argument struct's Free() function is modified to check if the last batch has been handled, and if not, it sends the last batch to the channel. Additionally, the channel in the JoinRun() function is modified to have a buffer size of `mcpu`.

There are no major issues with the changes made in the pull request. However, the title and body of the pull request could be more descriptive. The title could be more specific about what the fix is addressing, and the body could provide more context about the issue and why the changes were made. 

One suggestion for optimization is to add comments to the code to explain the changes made. This will make it easier for other developers to understand the code and make future modifications.

**Review by nnsgmsone — APPROVED** [review]

File: pkg/sql/colexec/right/types.go [hunk 1]
```diff
@@ -81,6 +83,15 @@ type Argument struct {
 func (arg *Argument) Free(proc *process.Process, pipelineFailed bool) {
 	ctr := arg.ctr
 	if ctr != nil {
+		if !ctr.handledLast {
+			if arg.IsMerger {
+				for i := uint64(1); i < arg.NumCPU; i++ {
```
Comment (m-schen) [inline pkg/sql/colexec/right/types.go:88]: 总觉得很诡异

**Review by m-schen — APPROVED** [review]

File: pkg/sql/colexec/right/types.go (hunk 1 above)
Comment (dongdongyang33) [reply pkg/sql/colexec/right/types.go:88]: 因为 right join 所在 pipeline 的父 pipeline 可能提前结束了，这个时候一组 right join 中有部分可能还没有走到 sendLast 的步骤就直接退出了（比如该 pipeline 的最后的发送节点发现想要发送的对象已经 context done 了，就会让该 pipeline 直接退出）
但是 sendLast 步骤里面有个阻塞等待的 channel，导致已经进入 sendLast 的 rightjoin 卡死在哪里

Comment (dongdongyang33) [reply pkg/sql/colexec/right/types.go:88]: 不过好像确实只要增加 channel 的长度就可以解决，不需要增加这个变量+排空 channel， 关于这一点我再思考一下

**Pull request merged**